### PR TITLE
Add MIT IPEA project dataset

### PR DESCRIPTION
**Pull request by fernandascovino**
- vw_h3: View which converts geometry column from STRING to GEOGRAPHY type and adds centriod column. To be used in multiple locations in the future.
- vw_vehicle_details: Basic view. Takes `rj-smtr.veiculo.sppo_licenciamento` and aliases column names to english.
- h3_gps: Assigns each GPS ping to a H3 tile using vw_h3. This consolidates multiple GPS pings for a given vehicle into a single h3 tile tile_entry_time and tile_exit_time.

### CODE REVIEW

File: models/mit_ipea_project/h3_gps.sql [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        data AS as_at,
+        hora AS time,
+        id_veiculo AS vehicle_id,
+        servico AS service,
+        latitude,
+        longitude,
+        ST_GEOGPOINT(longitude, latitude) AS geography,
+        tile_id
+    FROM brt_sppo_gps
+    LEFT JOIN `rj-smtr-dev.mit_ipea_project.vw_h3` AS h3
```
Comment (eng-rodrigocunha) [inline models/mit_ipea_project/h3_gps.sql:48]: Change to ref

File: models/mit_ipea_project/h3_gps.sql [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+h3_gps AS (
+    SELECT
+        *,
+        CASE
+          WHEN LAG(tile_id) OVER (ORDER BY vehicle_id, timestamp_gps) = tile_id THEN 0
+          ELSE 1
+        END AS tile_entry -- 1 when the vehicle first enters the tile. If 0, the vehicle is already in the tile.
+    FROM gps
+    LEFT JOIN `rj-smtr-dev.mit_ipea_project.vw_h3` AS h3
```
Comment (eng-rodrigocunha) [inline models/mit_ipea_project/h3_gps.sql:61]: Change to ref

File: models/mit_ipea_project/h3_gps.sql [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        modo, timestamp_gps, data, hora, id_veiculo, servico, latitude, longitude
+    FROM `rj-smtr.br_rj_riodejaneiro_veiculos.gps_sppo`
+    WHERE data = "{{ var('run_date') }}" -- This is a hardcoded date to allow for testing. This should be removed in production.
+
+    UNION DISTINCT
+
+    SELECT
+        modo, timestamp_gps, data, hora, id_veiculo, servico, latitude, longitude
+    FROM `rj-smtr.br_rj_riodejaneiro_veiculos.gps_brt`
+    WHERE data = "{{ var('run_date') }}" -- This is a hardcoded date to allow for testing. This should be removed in production.
```
Comment (ashleyvicary) [inline models/mit_ipea_project/h3_gps.sql:30]: Remove these comments

File: models/mit_ipea_project/h3_gps.sql [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    LEFT JOIN `rj-smtr-dev.mit_ipea_project.vw_h3` AS h3
+    ON ST_DWITHIN(ST_GEOGPOINT(longitude, latitude), h3.centroid, 560)
+
+),
+
+h3_gps AS (
+    SELECT
+        *,
+        CASE
+          WHEN LAG(tile_id) OVER (ORDER BY vehicle_id, timestamp_gps) = tile_id THEN 0
```
Comment (eng-rodrigocunha) [inline models/mit_ipea_project/h3_gps.sql:57]: WHEN LAG(tile_id) OVER (PARTITION BY vehicle_id ORDER BY timestamp_gps) = tile_id THEN 0

**Review by eng-rodrigocunha — APPROVED** [review]
LGTM

**Pull request merged**